### DISCUSSION TITLE
SMTP errror, no encryption: internal client error: No compatible authentication mechanism was found

### DISCUSSION THREAD

**Discussion by deraljoscha** [dani-garcia/vaultwarden#2607] [Q&A]
So I have the latest version of vaultwarden running on two servers. On one intance everythiung is running fine with SMTP over port 587 and starttls, on the other one I want to use a SMTP server which works with no encryption over port 25
my config looks like this:
SMTP_HOST=smtp.example.com
SMTP_FROM=mail@example.com
SMTP_FROM_NAME=Vaultwarden
SMTP_PORT=25
SMTP_SECURITY=off
SMTP_USERNAME=myusername
SMTP_PASSWORD=mypassword

I tried every option I could fin in the .env template
https://github.com/dani-garcia/vaultwarden/blob/main/.env.template

But I keep getting this error:

[vaultwarden::mail][ERROR] SMTP Client error: internal client error: No compatible authentication mechanism was found
[response][INFO] (reinvite_user) POST /api/organizations/<org_id>/users/<user_org>/reinvite => 400 Bad Request

I know my smtp genrally config works, I have it set up with the same options in other docker apps like uptime-kuma for example

would be grateful for some help
thanks in advance

**Comment by BlackDex** [comment]
Most mail servers do not allow login on port 25, and thus also do not send that information at all. This causes it to fall.

Either use port 587, or try without a login, so no password and username.

**Comment by deraljoscha** [comment]
As I mentioned, I know that my settings are correct. The mail server does allow login with these details on port 25
And it does look like an internal error of bitwarden/vaultwarden for me

**Reply by BlackDex** [reply]
Well, i suggest to do the following then.
Set the following environment variables.
* `LOG_LEVEL=Debug`
* `SMTP_DEBUG=true`

Restart Vaultwarden, and do a test again.
This can be done from the `/admin` interface under the SMTP settings.

Check the logs during the test.

I can assure you that Vaultwarden works with and without authentication for SMTP, so that should not be an issue.

Also, i suggest, that when possible, check the mail server logs also if you can, or contact the administrators of the mail server to see if they can provide some more information.

**Comment by deraljoscha** [comment]
Thanks for your help
I really hate these smtp servers of my company
Unfortunately I have no chance of getting the  the mail server logs, although I will ask the ip department for help.

So now this is the situation:

Same error still persists with port 25

```
[2022-07-09 14:31:46.108][lettre::transport::smtp::client::connection][DEBUG] << 250-smtp.example.com Hello 04e93d69c700 [my.server.ip]<CRLF>250-SIZE 52428800<CRLF>250-8BITMIME<CRLF>250-PIPELINING<CRLF>250-CHUNKING<CRLF>250-STARTTLS<CRLF>

[2022-07-09 14:31:46.108][lettre::transport::smtp::client::connection][DEBUG] << 250-smtp.example.com Hello 04e93d69c700 [my.server.ip]<CRLF>250-SIZE 52428800<CRLF>250-8BITMIME<CRLF>250-PIPELINING<CRLF>250-CHUNKING<CRLF>250-STARTTLS<CRLF>250 HELP<CRLF>

[2022-07-09 14:31:46.108][lettre::transport::smtp::client::connection][DEBUG] server smtp.example.com with {EightBitMime, StartTls}

[2022-07-09 14:31:46.109][vaultwarden::mail][DEBUG] SMTP Client error: lettre::transport::smtp::Error {

    kind: Client,

    source: "No compatible authentication mechanism was found",

}

[2022-07-09 14:31:46.109][vaultwarden::mail][ERROR] SMTP Client error: internal client error: No compatible authentication mechanism was found
```


Found one other config over port 587, with "force_tls" but I'm getting another error (literally foing crazy here)

```
> Error sending SMTP test email
> SMTP Connection error: error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:331:
```

**Reply by BlackDex** [reply]
Basically what i see here is that you need to enable starttls, the server sends it supports that, and most servers only permit login after starttls has been started. If that is started it will send which kind of authentication is supports, and the mail sending can continue.

**Reply by BlackDex** [reply]
So, try this:
```ini
  SMTP_PORT=25
  SMTP_SECURITY=starttls
  SMTP_USERNAME=<mail-address>
  SMTP_PASSWORD=<password>
```

Or this:
```ini
  SMTP_PORT=587
  SMTP_SECURITY=starttls
  SMTP_USERNAME=<mail-address>
  SMTP_PASSWORD=<password>
```

**Comment by deraljoscha** [comment]
Yes, had the same idea.
I tried both again

Both once with

```
Accept Invalid Certs (Know the risks!)
Default: false
Accept Invalid Hostnames (Know the risks!)
Default: false
```
and without.

  SMTP_PORT=25
  SMTP_SECURITY=starttls

```
> Error sending SMTP test email
> SMTP Client error: internal client error: No compatible authentication mechanism was found
```

  SMTP_PORT=587
  SMTP_SECURITY=starttls

```
Error sending SMTP test email
SMTP Connection error: Connection refused (os error 111)
```

I guess I need to get in touch with my IP department

**Comment by deraljoscha** [comment]
One question:
uptime-kuma, where my smtp config works with port 25 has an option "ignore tls error" which I activated, is there a similar option in vaultwarden?

**Reply by BlackDex** [reply]
Those are available.
https://github.com/dani-garcia/vaultwarden/blob/b64cf27038f04368af8f25aa80782d37471e6303/.env.template#L371-L380

**Comment by kauayrocha** [comment]
Hi guys,, i'm have the same problem in latest version (1.26.0) with the docker container, @deraljoscha  did you solve this problem?

**Comment by alfonsrv** [comment]
Trying to use Vaultwarden with host-based relay instead of traditional username + password auth also causes this issue. Whitelisted hosts can send mails without any authentication.

SMTP-debug log output:

```
vaultwarden  | [2022-12-03 12:08:40.692][lettre::transport::smtp::client::async_connection][DEBUG] << 220 mail.domain.tld Microsoft ESMTP MAIL Service ready at Sat, 3 Dec 2022 12:08:37 +0100<CRLF>
vaultwarden  | [2022-12-03 12:08:40.692][lettre::transport::smtp::client::async_connection][DEBUG] Wrote: EHLO vaultwarden.domain.tld<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>250-DSN<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>250-DSN<CRLF>250-ENHANCEDSTATUSCODES<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>250-DSN<CRLF>250-ENHANCEDSTATUSCODES<CRLF>250-STARTTLS<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>250-DSN<CRLF>250-ENHANCEDSTATUSCODES<CRLF>250-STARTTLS<CRLF>250-8BITMIME<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>250-DSN<CRLF>250-ENHANCEDSTATUSCODES<CRLF>250-STARTTLS<CRLF>250-8BITMIME<CRLF>250-BINARYMIME<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] << 250-mail.domain.tld Hello [10.10.3.8]<CRLF>250-SIZE 36700160<CRLF>250-PIPELINING<CRLF>250-DSN<CRLF>250-ENHANCEDSTATUSCODES<CRLF>250-STARTTLS<CRLF>250-8BITMIME<CRLF>250-BINARYMIME<CRLF>250 CHUNKING<CRLF>
vaultwarden  | [2022-12-03 12:08:40.703][lettre::transport::smtp::client::async_connection][DEBUG] server mail.domain.tld with {EightBitMime, StartTls}
vaultwarden  | [2022-12-03 12:08:40.704][vaultwarden::mail][ERROR] SMTP Client error: internal client error: No compatible authentication mechanism was found
```

**Reply by BlackDex** [reply]
In that case, remove username and password, that way it will not look for login options.

**Reply by alfonsrv** [reply]
That's what I tried, using `SMTP_USERNAME=` and `SMTP_PASSWORD=` in `.env`.

If I comment them out completely I get the following error and the container won't start up.

```
vaultwarden  | Error loading config:
vaultwarden  |  Both `SMTP_USERNAME` and `SMTP_PASSWORD` need to be set to enable email authentication
```

Attached a screenshot of the configuration for your reference. Already tried setting `SMTP Auth mechanism` to both `plain` and `login` explicitly, as well as specifying the `From Address` in the `Username` field in case there's some weird quirk.

![vaultwarden](https://user-images.githubusercontent.com/48770755/205438993-3352618d-a88e-4632-a748-cf434fd0bf9e.png)

---

Running latest Vaultwarden `1.26.0`

**Reply by BlackDex** [reply]
@alfonsrv Then there actually is something set. Either username or password is set to something.
Check the **Support String** which you can generate via the Diagnostics page.
I just tested it with a local MailDev instance, and that works just fine.

**Reply by alfonsrv** [reply]
After explicitly adding `SMTP_AUTH_MECHANISM=` to my `.env` it works as expected. Thanks!

`"smtp_password": "***",` was indeed set – though environment variable was empty and nothing was set in the GUI (also checked for invisible characters). After setting the `SMTP_AUTH_MECHANISM` it shows `"smtp_password": null,`.
